src/google-apis/auth-client.ts

```diff
@@ -18,7 +18,8 @@
 import * as common from '@google-cloud/common';
 const pkg = require('../../../package.json');
 import * as is from 'is';
-var isFunction = is.fn;
+// TODO: Address the error where `is` does not have a `fn` property
```

Comment:
Isn't it better to throw an error when `is.fn` doesn't exist?
```typescript
if (!is.fn) {
  throw new Error('is.fn is not defined');
}
var isFuntion = is.fn;
```

Same for similar places in this PR.

Reply:
I think that is a good idea, but for these PRs, as much as I can, I don't want to make behavioral changes.  I will address the TODOs in follow-up PRs.